### PR TITLE
Add basic docker-dind test

### PR DESCRIPTION
**Pull request by tianon**

### CODE REVIEW

File: test/tests/docker-dind/run.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		false
+	fi
+	sleep 2
+done
+
+docker_ pull busybox
+
+docker_ run --rm busybox true
+
+docker_ create -i --name test busybox cat
```
Comment (md5) [inline test/tests/docker-dind/run.sh]: The name `test` seems a bit collision prone.

Comment (tianon) [inline test/tests/docker-dind/run.sh]: This is inside our container, so collision here should actually be
impossible.

Comment (md5) [inline test/tests/docker-dind/run.sh]: :+1:

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**